packages/core/src/client.ts

```diff
@@ -2222,13 +2225,13 @@ export class MedplumClient extends EventTarget {
     contentType?: string,
     options?: RequestInit
   ): Promise<any> {
-    let url;
+    let url: string | URL;
     if (typeof idOrIdentifier === 'string') {
       const id = idOrIdentifier;
       url = this.fhirUrl('Bot', id, '$execute');
     } else {
       const identifier = idOrIdentifier;
-      url = this.fhirUrl('Bot', '$execute') + `?identifier=${identifier.system}|${identifier.value}`;
+      url = this.fhirUrl('Bot', '$execute').toString() + `?identifier=${identifier.system}|${identifier.value}`;
```

Comment:
Good catch.

We should probably use the `URL` for this rather than concatenation query params due to escaping concerns.

I.e..,

```ts
      url = this.fhirUrl('Bot', '$execute');
      url.searchParams.set('identifier', identifier.system + '|' + identifier.value);
```